opennms-provision/opennms-detector-simple/src/main/java/org/opennms/netmgt/provision/detector/snmp/SnmpDetector.java

```diff
@@ -220,6 +221,27 @@ public boolean isServiceDetected(InetAddress address, SnmpAgentConfig agentConfi
         }
     }
 
+    @Override
+    public List<SnmpAgentConfig> getListOfAgentConfigs(DetectRequest request) {
+        List<SnmpAgentConfig> agentConfigList = new ArrayList<>();
+        Map<String, String> runTimeAttributes = request.getRuntimeAttributes();
+        if (hasMultipleAgentConfigs(runTimeAttributes)) {
+            runTimeAttributes.remove(HAS_MULTIPLE_AGENT_CONFIGS);
+            //Rest of the attribute values are snmp configs.
+            runTimeAttributes.forEach((label, configAsString) -> {
```

Comment:
Can we prefix the keys in the `runTimeAttributes` map with something like `SNMP_AGENT_CONFIG_` and only derive the agent configs from keys matching this prefix?

This would be a bit safer than assuming everything else is an agent config.